varya/classes/class-varya-custom-colors.php

```diff
@@ -0,0 +1,173 @@
+
+	function __construct() {
+
+		/**
+		 * Define color variables
+		 */
+		$this->$varya_custom_color_variables = array(
+			array( '--global--color-primary', '#000000', 'Primary Color' ),
+			array( '--global--color-secondary', '#A36265', 'Secondary Color' ),
+			array( '--global--color-foreground', '#333333', 'Foreground Color' ),
+			array( '--global--color-background-light', '#FAFBF6', 'Background Light Color' ),
+			array( '--global--color-background', '#FFFFFF', 'Background Color' ),
+			array( '--global--color-border', '#EFEFEF', 'Borders Color' )
+		);
+
+		/**
+		 * Register Customizer actions
+		 */
+		add_action( 'customize_register', array( $this, 'varya_customize_custom_colors_register' ) );
+
+		/**
+		 * Enqueue color variables for customizer & frontend.
+		 */
+		add_action( 'wp_enqueue_scripts', array( $this, 'varya_custom_color_variables' ) );
+
+		/**
+		 * Enqueue color variables for editor.
+		 */
+		add_action( 'enqueue_block_editor_assets', array( $this, 'varya_editor_custom_color_variables' ) );
+	}
+
+	/**
+	 * Add Theme Options for the Customizer.
+	 *
+	 * @param WP_Customize_Manager $wp_customize Theme Customizer object.
+	 */
+	function varya_customize_custom_colors_register( $wp_customize ) {
+
+		/**
+		 * Create color options panel.
+		 */
+		$wp_customize->add_section( 'varya_options', array(
+			'capability' => 'edit_theme_options',
+			'title'      => esc_html__( 'Colors', 'varya' ),
+		) );
+
+		/**
+		 * Create toggle between default and custom colors.
+		 */
+		$wp_customize->add_setting(
+			'custom_colors_active',
+			array(
+				'capability'        => 'edit_theme_options',
+				'sanitize_callback' => array( $this, 'sanitize_select' ),
+				'transport'         => 'refresh',
+				'default'           => 'default',
+			)
+		);
+
+		$wp_customize->add_control(
+			'custom_colors_active',
+			array(
+				'type'    => 'radio',
+				'section' => 'varya_options',
+				'label'   => __( 'Colors', 'varya' ),
+				'choices' => array(
+					'default' => __( 'Theme Default', 'varya' ),
+					'custom'  => __( 'Custom', 'varya' ),
+				),
+			)
+		);
+
+		/**
+		 * Create customizer color controls.
+		 */
+		foreach ( $this->$varya_custom_color_variables as $variable ) {
+			$wp_customize->add_setting(
+				"varya_$variable[0]",
+				array(
+					'default'	=> esc_html( $variable[1] )
+				)
+			);
+			$wp_customize->add_control( new WP_Customize_Color_Control(
+				$wp_customize,
+				"varya_$variable[0]",
+				array(
+					'section'   => 'varya_options',
+					'label'     => __( $variable[2], 'varya' ),
+					'active_callback' => function() use ( $wp_customize ) {
+						return ( 'custom' === $wp_customize->get_setting( 'custom_colors_active' )->value() );
+					},
+				)
+			) );
+		}
+	}
+
+	/**
+	 * Generate color variables for customizer & frontend.
+	 */
+	function varya_generate_custom_color_variables( $context = null ) {
+
+		if ( $context === 'editor' ) {
+			$theme_css = ':root .editor-styles-wrapper {';
```

Comment:
Is it worth inserting a comment here about why this is necessary? Asking because Gutenberg was trying to move away from having to target this class, IIRC.

Reply:
On second thought, it's pretty straightforward from just reading the if statement.